lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -3812,25 +3805,31 @@ PotentialArchetype *GenericSignatureBuilder::realizePotentialArchetype(
   return pa;
 }
 
+static Type getStructuralType(TypeDecl *typeDecl, LazyResolver *resolver) {
```

Comment:
I'd drop the `LazyResolver` parameter here. We can get it from `typeDecl->getASTContext().getLazyResolver()` in the few cases where we need it.

Reply:
I think @xymus can do that next, after landing the getStructuralType() request :)